imgui_plot.h

```diff
@@ -1102,14 +1105,32 @@ namespace ImGui
             }
             float const * const y = y_data_bottoms ? y_data_bottoms : y_data;
 
+            std::vector<ImVec2> v;
+
             ImDrawList* dl = window->DrawList;
-            dl->PathLineTo(ImVec2(figure.GetScalesX()->Scale(x_data[0]), figure.GetScalesY()->Scale(y[0] - y_data[0])));
+            v.emplace_back(figure.GetScalesX()->Scale(x_data[0]), figure.GetScalesY()->Scale(y[0] - y_data[0]));
+            dl->PathLineTo(v.back());
 
             for (size_t i = 0; i < n_data; ++i) {
-                dl->PathLineTo(ImVec2(figure.GetScalesX()->Scale(x_data[i]), figure.GetScalesY()->Scale(y[i])));
+                v.emplace_back(figure.GetScalesX()->Scale(x_data[i]), figure.GetScalesY()->Scale(y[i]));
+                dl->PathLineTo(v.back());
+            }
+
+            v.emplace_back(figure.GetScalesX()->Scale(x_data[n_data-1]), figure.GetScalesY()->Scale(y[n_data-1] - y_data[n_data-1]));
+            dl->PathLineTo(v.back());
+
+            ImVec2 pointer(GImGui->IO.MousePos.x, GImGui->IO.MousePos.y);
+
+            const bool check = is_inside_area(pointer, v);
```

Comment:
I like this idea.  Having a function that checks if the mouse goes inside of a polygon could simplify the code I had written previously, and also make life much easier for the more complicated plots.